website/project/tasks.py

```diff
@@ -38,6 +38,26 @@ def on_node_updated(node_id, user_id, first_save, saved_fields, request_headers=
     if node.get_identifier_value('doi') and bool(node.IDENTIFIER_UPDATE_FIELDS.intersection(saved_fields)):
         node.request_identifier_update(category='doi')
 
+@celery_app.task(ignore_results=True)
+def update_files_count(node_id):
+    AbstractNode = apps.get_model('osf.AbstractNode')
+    node = AbstractNode.load(node_id)
+
+    if node is None:
+        return
+
+    field = AbstractNode._meta.get_field('modified')
+    # Do not update modified upon save()
+    field.auto_now = False
+
+    if node.is_registration:
+        node.files_count = node.files.filter(deleted_on__isnull=True).count()
+    # else:
+        # TODO: Do node specific files_count update
```

Comment:
If we're planning to do this for nodes, why is the filed only added to registrations?

Reply:
With `AbstractNode` being a `TypedModel`, the schema migration sets the `files_count` for both nodes and registrations to `null`. Then the data migration part computes the actual counts for successful registrations.

Reply:
Sure. The task was used in the hopes that we would need this count for nodes as well.  It seemed like a good idea to handle both cases in one place.